Explorer/Assets/DCL/Navmap/SearchResultPanelController.cs

```diff
@@ -0,0 +1,80 @@
+using Cysharp.Threading.Tasks;
+using DCL.AssetsProvision;
+using DCL.PlacesAPIService;
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using UnityEngine.Pool;
+using Object = UnityEngine.Object;
+
+namespace DCL.Navmap
+{
+    public class SearchResultPanelController
+    {
+        public event Action<string> OnResultClicked;
+
+        private readonly SearchResultPanelView view;
+        private readonly IAssetsProvisioner assetsProvisioner;
+        private ObjectPool<FullSearchResultsView> resultsPool;
+        private readonly List<FullSearchResultsView> usedPoolElements;
+
+        public SearchResultPanelController(SearchResultPanelView view, IAssetsProvisioner assetsProvisioner)
+        {
+            this.view = view;
+            this.assetsProvisioner = assetsProvisioner;
+            usedPoolElements = new List<FullSearchResultsView>();
+            InitPoolAsync().Forget();
+        }
+
+        private async UniTaskVoid InitPoolAsync()

```

Comment:
I think it should not be here: all initialization like this and resolving assets references should be done in a corresponding plugin